### PR TITLE
[Fix] Fix lint

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers. By the way, if you're not familiar with how to use pre-commit to fix lint issues or add unit tests, please refer to [Contributing to OpenMMLab](https://mmengine.readthedocs.io/en/latest/notes/contributing.html).

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.](https://github.com/open-mmlab/mmcv/pull/3196)

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMPretrain.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: .pre-commit-config-zh-cn.yaml [hunk 1]
```diff
@@ -55,11 +59,12 @@ repos:
         args: ["mmengine", "tests"]
       - id: remove-improper-eol-in-cn-docs
   - repo: https://gitee.com/openmmlab/mirrors-mypy
-    rev: v0.812
+    rev: v1.2.0
```
Comment (zhouzaida) [inline .pre-commit-config-zh-cn.yaml:62]: 这个 repo 下是不是没有这个分支

File: .pre-commit-config-zh-cn.yaml [hunk 2]
```diff
@@ -13,7 +17,7 @@ repos:
     hooks:
       - id: yapf
   - repo: https://gitee.com/openmmlab/mirrors-pre-commit-hooks
-    rev: v4.3.0
+    rev: v5.0.0
```
Comment (zhouzaida) [inline .pre-commit-config-zh-cn.yaml:20]: 这个 repo 下是不是没有这个分支

File: mmengine/dist/utils.py [hunk 3]
```diff
@@ -563,7 +563,7 @@ def cast_data_device(
         Tensor or list or dict: ``data`` was casted to ``device``.
     """
     if out is not None:
-        if type(data) != type(out):
+        if type(data) is type(out):
```
Comment (zhouzaida) [inline mmengine/dist/utils.py:566]:
```suggestion
        if type(data) is not type(out):
```

File: .pre-commit-config-zh-cn.yaml (hunk 1 above)
Comment (HAOCHENYE) [reply .pre-commit-config-zh-cn.yaml:62]: 没有，要是有人用的话再去同步一下好了。我貌似没有权限

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**